src/molecular-weight.jl

```diff
@@ -0,0 +1,260 @@
+const AA_Z_MW_MW  = (AA_Q_MW + AA_D_MW)/2     # Glutamine or Glutamine      ->  Average between Glutamine and Glutamine
+const AA_X_MW_MW  = 110                       # Any amino acid (average)    ->  Source = http://thermofisher.com/de/de/home/references/ambion-tech-support/rna-tools-and-calculators/proteins-and-amino-acids.html
+const AA_O_Gap_MW = 0                         # Consider gap as 0       
+
+# Defining the function molecular_weight, which accepts a AA sequence and units ("gmol", "KDa") as input and return molecular weight in either g/mol or kDa. If not specified it returns in g/mol
+function molecular_weight(aa_seq::AASeq, unit=:"gmol")
+    AA_vector = Vector{Float64}(undef, 0)
+    i = 1
+    while i <= length(aa_seq)
+        if aa_seq[i] == AA_A
+            push!(AA_vector, AA_A_MW)
+        elseif aa_seq[i] == AA_R
+            push!(AA_vector, AA_R_MW)
+        elseif aa_seq[i] == AA_N
+            push!(AA_vector, AA_N_MW)
+        elseif aa_seq[i] == AA_D
+            push!(AA_vector, AA_D_MW)
+        elseif aa_seq[i] == AA_C
+            push!(AA_vector, AA_C_MW)
+        elseif aa_seq[i] == AA_Q
+            push!(AA_vector, AA_Q_MW)
+        elseif aa_seq[i] == AA_E
+            push!(AA_vector, AA_E_MW)
+        elseif aa_seq[i] == AA_G
+            push!(AA_vector, AA_G_MW)
+        elseif aa_seq[i] == AA_H
+            push!(AA_vector, AA_H_MW)  
+        elseif aa_seq[i] == AA_I
+            push!(AA_vector, AA_I_MW)
+        elseif aa_seq[i] == AA_L
+            push!(AA_vector, AA_L_MW)    
+        elseif aa_seq[i] == AA_K
+            push!(AA_vector, AA_K_MW)
+        elseif aa_seq[i] == AA_M
+            push!(AA_vector, AA_M_MW)                                            
+        elseif aa_seq[i] == AA_F
+            push!(AA_vector, AA_F_MW)
+        elseif aa_seq[i] == AA_P
+            push!(AA_vector, AA_P_MW)
+        elseif aa_seq[i] == AA_S
+            push!(AA_vector, AA_S_MW)
+        elseif aa_seq[i] == AA_T
+            push!(AA_vector, AA_T_MW)   
+        elseif aa_seq[i] == AA_W
+            push!(AA_vector, AA_W_MW)
+        elseif aa_seq[i] == AA_Y
+            push!(AA_vector, AA_Y_MW)
+        elseif aa_seq[i] == AA_V
+            push!(AA_vector, AA_V_MW)
+        elseif aa_seq[i] == AA_O
+            push!(AA_vector, AA_O_MW_MW)
+        elseif aa_seq[i] == AA_U
+            push!(AA_vector, AA_U_MW_MW)
+        elseif aa_seq[i] == AA_B
+            push!(AA_vector, AA_B_MW_MW)
+        elseif aa_seq[i] == AA_J
+            push!(AA_vector, AA_J_MW_MW)
+        elseif aa_seq[i] == AA_Z
+            push!(AA_vector, AA_Z_MW_MW)
+        elseif aa_seq[i] == AA_X
+            push!(AA_vector, AA_X_MW_MW)
+        elseif aa_seq[i] == AA_O_Gap
+            push!(AA_vector, AA_O_Gap_MW)                           
+        else
+            println("Error due to unknown amino acid at $i position")
+            return nothing
```

Comment:
It's generally better to throw errors than to print them. The reason is that
a) printing only really works if you have a human in the loop which is paying attention. It doesn't work well if you want to run this function using a program, and then programmatically handle the error state.
b) Although unlikely, the value `nothing` might not throw an error where it is used. So there is a risk the program may continue even though it should have errored. For example, if you do:

```
julia> print(molecular_weight(aa_seq))
Error due to unknown amino acid at $i position
nothing
```